packages/xml-views-validation/src/validators/elements/non-stable-id.ts

```diff
@@ -0,0 +1,123 @@
+import { some, includes } from "lodash";
+import { XMLElement } from "@xml-tools/ast";
+import {
+  getUI5ClassByXMLElement,
+  ui5NodeToFQN,
+  resolveXMLNS,
+} from "@ui5-language-assistant/logic-utils";
+import { UI5SemanticModel } from "@ui5-language-assistant/semantic-model-types";
+import { NonStableIDIssue } from "../../../api";
+import { NON_STABLE_ID, getMessage } from "../../utils/messages";
+import { isCustomClass } from "../../utils/custom-class";
+import { CORE_NS } from "../../utils/special-namespaces";
+
+export function validateNonStableId(
+  xmlElement: XMLElement,
+  model: UI5SemanticModel
+): NonStableIDIssue[] {
+  // Can't give an error if there is no position or name
+  if (xmlElement.name === null || xmlElement.syntax.openName === undefined) {
+    return [];
+  }
+
+  if (isWhiteListed(xmlElement, model)) {
+    return [];
+  }
+
+  if (!isCustomClass(xmlElement)) {
+    return [];
+  }
+
+  if (
+    hasNonAdaptableMetaData(xmlElement) ||
+    hasNonAdaptableTreeMetaData(xmlElement)
+  ) {
+    return [];
+  }
+
+  if (isElementWithStableID(xmlElement)) {
+    return [];
+  }
+
+  const nonStableIDIssue: NonStableIDIssue = {
+    kind: "NonStableIDIssue",
+    message: getMessage(NON_STABLE_ID, xmlElement.name),
+    severity: "error",
+    offsetRange: {
+      start: xmlElement.syntax.openName.startOffset,
+      end: xmlElement.syntax.openName.endOffset,
+    },
+  };
+
+  return [nonStableIDIssue];
+}
+
+function isWhiteListed(
```

Comment:
- Do we need to use two APIs for checking the names?
  - `ui5NodeToFQN` vs resolveNS + includes
  - could/should we use only the resolve + includes for consistency?

- Do we need to distinguish between things a the root or not?
  - because `["Fragment", "CustomData", "ExtensionPoint"]` would still return true if they are in the root if I read this correctly.

So maybe this function could be refactored, maybe even simplified greatly.